test/e2e/cmd/run/job.go

```diff
@@ -49,29 +52,18 @@ func NewJob(podName, templatePath string, writer io.Writer, timestampExtractor t
 		runningWg:          runningWg,
 		logStreamWg:        logStreamWg,
 		timestampExtractor: timestampExtractor,
+		resultFile:         resultFile,
 	}
 }
 
-// WaitForLogs waits for logs to be fully read before leaving.
-func (j *Job) WaitForLogs() {
-	if j.stopRequested {
-		// already done in the past
-		return
-	}
-	j.stopRequested = true
-	close(j.stopLogStream)
-	log.Info("Waiting for log stream to be over", "name", j.jobName)
-	j.logStreamWg.Wait()
-	close(j.streamErrors)
-}
-
 // Stop is only a best effort to stop the streaming process
 func (j *Job) Stop() {
 	if j.stopRequested {
 		// Job already stopped
 		return
 	}
 	close(j.stopLogStream)
+	close(j.streamErrors)
```

Comment:
I'm not sure why we didn't close this channel.

Reply:
I'll remove `stopRequested`.

What confused me is that:
In `WaitForLogs()`, we closed the 2 channels `stopLogStream` and `streamErrors`.
In `Stop()`, we closed only channel `stopLogStream`.
We almost always called `WaitForLogs()` and then `Stop()` except on pod deletion.